prerequisites/install-functions/build_opencoarrays.sh

```diff
@@ -1,3 +1,23 @@
+work_around_cmake_issue()
+{
+  # This function ensures that the desired mpiexec is appears before any other mpiexec in the PATH of the process that runs
+  # cmake in the build_opencoarrays function below.
+   
+  # This is a (hopefully temporary) workaround for CMake issue 16804 (https://gitlab.kitware.com/cmake/cmake/issues/16804):
+  # even when MPI_C_COMPILER and MPI_Fortran_COMPILER are set as they are in the $CMAKE invocation below, cmake finds the
+  # mpiexec program that is first in the PATH or fails to find any mpiexec if it's not in the PATH.
+
+  MPIEXEC_PATH=${MPIFC%/*}
+  if [[ ! -x ${MPIEXEC_PATH}/mpiexec ]]; then
+    emergency "work_around_cmake_issue: executable mpiexec not found in ${MPIEXEC_PATH}"
+  fi
+  if [[ -z "${PATH}" ]]; then
+    export PATH="${MPIEXEC_PATH}"
+  else
+    export PATH="${MPIEXEC_PATH}":"${PATH}"
+  fi
+}
```

Comment:
I always forget the precise semantics of variable scoping in bash, but it might be nice to store the user's old PATH before updating it, and then return it to its original state... This may in fact already be the case due to variable scoping, unless, of course, the user sources `build_opencoarrays.sh` rather than running it directly, since, in that case, no subprocess will be created.